iceoryx_hoofs/legacy/include/iceoryx_hoofs/cxx/filesystem.hpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_CXX_FILESYSTEM_HPP
+#define IOX_HOOFS_CXX_FILESYSTEM_HPP
+
+#include "iox/filesystem.hpp"
+
+namespace iox
+{
+/// @todo iox-#1593 Deprecate include
+/// [[deprecated("Deprecated in 3.0, removed in 4.0, please include 'iox/filesystem.hpp' instead")]]
+namespace cxx
+{
+namespace internal
+{
+/// @deprecated use `iox::internal::ASCII_A` instead of `iox::cxx::internal::ASCII_A`
+using iox::internal::ASCII_A;
+/// @deprecated use `iox::internal::ASCII_Z` instead of `iox::cxx::internal::ASCII_Z`
+using iox::internal::ASCII_Z;
+/// @deprecated use `iox::internal::ASCII_CAPITAL_A` instead of `iox::cxx::internal::ASCII_CAPITAL_A`
+using iox::internal::ASCII_CAPITAL_A;
```

Comment:
Could you check if this was part of the v2 release? If not, the header could be removed.

Reply:
Most of this file was part of the v2 release but was located in the helplets file. @mossmaurice will add a helplets legacy header (https://github.com/eclipse-iceoryx/iceoryx/pull/1858#issuecomment-1409442722). Therefore, I removed everything from the filesystem legacy header except the perms enum class.